src/main/java/wooteco/subway/service/StationService.java

```diff
@@ -0,0 +1,44 @@
+package wooteco.subway.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import wooteco.subway.dao.StationDao;
+import wooteco.subway.domain.Station;
+import wooteco.subway.exception.DataNotExistException;
+
+@Service
+public class StationService {
+
+    private final StationDao stationDao;
+
+    public StationService(StationDao stationDao) {
+        this.stationDao = stationDao;
+    }
+
+    public Station createStation(Station station) {
+        Optional<Station> foundStation = stationDao.findByName(station.getName());
+        if (foundStation.isPresent()) {
+            throw new IllegalArgumentException("이미 등록된 역입니다.");
+        }
+        return stationDao.save(station);
+    }
+
+    public Station findById(Long id) {
+        Optional<Station> foundStation = stationDao.findById(id);
+        if (foundStation.isEmpty()) {
+            throw new DataNotExistException("존재하지 않는 역입니다.");
+        }
+        return foundStation.get();
+    }
```

Comment:
`Optional`을 잘 사용하면 조금 더 간결한 코드로 같은 일을 할 수 있을 것 같아요~

Reply:
```suggestion
    public Station findById(Long id) {
        Optional<Station> foundStation = stationDao.findById(id);
        return foundStation.orElseThrow(() -> new DataNotExistException("존재하지 않는 역입니다."));
    }
```
`Optional` 이 제공하는 메서드를 사용해서 위와 같이 변경해보았습니다!!!! 감사합니다👻👻